### PR TITLE
Fix syncing time

### PR DESCRIPTION
**Pull request by amotl**
Dear Dave,

within #124, we found that the "sync_time" method would set the "tm_wday" slot within the timedata struct erroneous, at least on Pycom MicroPython.

While we have neither any generic ESP32 boards on our workbench, nor any STM32 boards at hand, we hope this fix will also be correct for Genuine MicroPython.

With kind regards,
Andreas.

### CODE REVIEW

File: rshell/main.py [hunk 1]
```diff
@@ -932,6 +932,7 @@ def rsync(src_dir, dst_dir, mirror, dry_run, print_func, recursed, sync_hidden):
                       "'{}' is a directory. Ignoring"
                 print_err(msg.format(src_filename, dst_filename))
             else:
+                print_func('Checking {}'.format(dst_filename))
```
Comment (dhylands) [inline rshell/main.py]: This line doesn't seem to be related to the PR

File: rshell/main.py [hunk 2]
```diff
@@ -942,20 +943,27 @@ def rsync(src_dir, dst_dir, mirror, dry_run, print_func, recursed, sync_hidden):
 def set_time(rtc_time):
     rtc = None
     try:
-        # Pyboard (pyboard doesn't have machine.RTC())
+        # Pyboard (pyboard doesn't have machine.RTC()).
+        # The pyb.RTC.datetime function takes the arguments in the order:
+        # (year, month, day, weekday, hour, minute, second, subseconds)
+        # http://docs.micropython.org/en/latest/library/pyb.RTC.html#pyb.RTC.datetime
```
Comment (dhylands) [inline rshell/main.py]: Great - thanks for adding the comments. I meant to suggest doing that, and I see you've done it.

File: rshell/main.py (hunk 1 above)
Comment (amotl) [reply rshell/main.py]: That's true. I found it convenient in the context of this change request to give the user some feedback what is going on. We are syncing the pretty large codebase of the [Terkin-Datalogger](https://github.com/hiveeyes/terkin-datalogger) which will take some time to upload. Even if no files have been changed, it will take some time.

Please let me know if you accept this or whether you want me to remove this line.

Comment (davehylands) [reply rshell/main.py]: It's not relevant to this particular PR, so it should be removed. It's probably worth adding, but should be done in a separate PR.

Comment (amotl) [reply rshell/main.py]: All right. I've removed the line as requested. Thanks again!

**Comment by dhylands** [comment]
I took your changes and added some of my own in #129 

Could you verify that #129 works for you?

Thanks

**Comment by amotl** [comment]
> Could you verify that #129 works for you?

It does. Thanks again!
